File: JAVA/Remove all occurences of duplicates in a linked list.java
class Solution {
    public Node removeAllDuplicates(Node head) {
        Node dummy = new Node(0);
        dummy.next = head;
        Node prev2 = dummy;
        Node curr = dummy;
        curr = curr.next;
        while(curr != null && curr.next != null) {
            boolean d = false;
            while(curr != null && curr.next != null && 
                    curr.data == curr.next.data) {
                curr = curr.next;
                d = true;
            }
            if(d) {
                prev2.next = curr.next;
                curr = curr.next;
            }
            else {
                prev2 = curr;
                curr = curr.next;
            }
        }        
        return dummy.next;
    }
}

File: C++/Remove all occurences of duplicates in a linked list.cpp
class Solution {
  public:
    Node* removeAllDuplicates(struct Node* head) {
        if(head == NULL) return NULL;
        if(head->next == NULL) return head;
        Node *prevnode = head;
        Node *currnode = head;
        while(currnode != NULL && currnode->next != NULL)
        {
            Node *nextnode = currnode->next;
            if(currnode->data != nextnode->data)
            {
                prevnode = currnode;
                currnode = nextnode;
            }
            else
            {   
                int val = currnode->data;
                if(currnode == head)
                {
                    while(head!=NULL && head->data == val)
                    {
                        head = head->next;
                    }
                    currnode = head;
                    prevnode = head;
                }
                else
                {
                    while(currnode != NULL && currnode->data == val)
                    {
                        currnode = currnode->next;
                    }
                    prevnode->next = currnode;
                }
            }
        }
        return head;
    }
};
